app/src/main/kotlin/com/vestrel00/daggerbutterknifemvp/ui/common/view/BaseFragment.kt

```diff
@@ -0,0 +1,163 @@
+import dagger.android.DispatchingAndroidInjector
+import dagger.android.HasFragmentInjector
+import javax.inject.Inject
+
+/**
+ * Abstract (Dialog)Fragment for all (Dialog)Fragments and child (Dialog)Fragments to extend.
+ * This contains some boilerplate dependency injection code and activity [Context].
+ *
+ * **WHY EXTEND DialogFragment?**
+ *
+ * [DialogFragment]s are simple extensions of Fragments. DialogFragments can be shown as a
+ * dialog floating above the current activity or be embedded into views like regular fragments.
+ * Therefore, supporting both Fragments and DialogFragments for dependency injection can simply be
+ * achieved by having the base fragment class (this) extend DialogFragment instead of Fragment.
+ * We could have separate base classes for Fragments and DialogFragments but that would produce
+ * duplicate code. See See https://github.com/vestrel00/android-dagger-butterknife-mvp/pull/64
+ *
+ * Note that as of Dagger 2.12, the abstract base framework type
+ * [dagger.android.DaggerDialogFragment] has been introduced for subclassing if so desired.
+ *
+ * **DEPENDENCY INJECTION**
+ *
+ * We could extend [dagger.android.DaggerDialogFragment] so we can get the boilerplate dagger code
+ * for free. However, we want to avoid inheritance (if possible and it is in this case) so that we
+ * have to option to inherit from something else later on if needed.
+ *
+ * **VIEW BINDING**
+ *
+ * This fragment handles view bind and unbinding.
+ */
+abstract class BaseFragment : DialogFragment(), HasFragmentInjector {
+
+    @Inject
+    lateinit var childFragmentInjector: DispatchingAndroidInjector<Fragment>
+
+    /*
+     * The property `childFragmentManager` cannot be declared here with that name as it results in
+     * an 'accidental override' compile-time error. Kotlin automatically generates getters for
+     * properties. In this case, Kotlin generates a method called `getChildFragmentManager()` for
+     * the property `childFragmentManager`, which then conflicts with the Activity's
+     * `getChildFragmentManager()` method.
+     *
+     * There are workarounds to this;
+     *
+     * 1. Change the property name to something else such as `fragmentChildFragmentManager`.
+     * 2. Rename the getter generated by Kotlin using @get:JvmName("fragmentChildFragmentManager")
+     *
+     * However, we will not be doing any of the above because subclasses would have to use our
+     * custom getter/property instead of the builtin `childFragmentManager`, which may cause
+     * confusion.
+     *
+     * The caveat to this is that we would need to use Robolectric in our tests in order to mock and
+     * verify `FragmentManager` usages. For more details around testing `FragmentManager` usages,
+     * see https://github.com/vestrel00/android-dagger-butterknife-mvp/pull/52.
+     */
+    // @Inject
+    // @field:Named(BaseFragmentModule.CHILD_FRAGMENT_MANAGER)
+    // protected lateinit var childFragmentManager: FragmentManager
```

Comment:
We don't inject our own reference of the `FragmentManager` here. Read the above block comment for more information.